### PR TITLE
Crosscompile for .Net Core 2.1 and .Net Core 3.0

### PR DESCRIPTION
**Pull request by Maxwe11**
### :sparkles: What kind of change does this PR introduce? (Bug fix, feature, docs update...)
Support of .Net Core 3.0

### :arrow_heading_down: What is the current behavior?
Described in #876 

### :thinking: Checklist before submitting

- [ ] All projects build
- [ ] Relevant documentation was updated

### CODE REVIEW

**Comment by skolima** [comment]
I'm sorry @Maxwe11 but I'm afraid I no longer use NuKeeper.

**Comment by AnthonySteele** [comment]
I can have a look over the weekend, thanks.

**Comment by AnthonySteele** [comment]
The code changes seem fine, is the build failing because of the version change?

**Comment by Maxwe11** [comment]
@AnthonySteele yeah, something wrong with CI infrastructure, I'd appreciate assistance

**Comment by AnthonySteele** [comment]
We need to change this line: https://github.com/NuKeeperDotNet/NuKeeper/blob/master/.travis.yml#L4

**Comment by AnthonySteele** [comment]
Same error in both Travis and ADPO pipeline:
```
/opt/hostedtoolcache/dotnet/sdk/3.1.101/Sdks/Microsoft.NET.Sdk/targets/Microsoft.PackageDependencyResolution.targets(234,5): 
error NETSDK1005: 
Assets file '/home/vsts/work/1/s/NuKeeper.AzureDevOps/obj/project.assets.json' 
doesn't have a target for '.NETStandard,Version=v2.0'. 
Ensure that restore has run and that you have included 'netstandard2.0' in the TargetFrameworks for your project. 
[/home/vsts/work/1/s/NuKeeper.AzureDevOps/NuKeeper.AzureDevOps.csproj]

```

This is a weird one, I have not seen it before. But since it happens in both places, it doesn't appear to be a quirk or travis.

**Comment by slang25** [comment]
It's weird for sure, I'm taking a look cause I'm interested. This kind of thing can happen when the obj folder is stale and the assets file has been created for different frameworks, but that doesn't appear to be what's happening here at a quick glance.

**Comment by slang25** [comment]
`Directory.Build.props`! 🤦‍♂️

**Comment by slang25** [comment]
We should take `TargetFrameworks` out of the `Directory.Build.props` file, as it needs to vary across multiple projects.

**Comment by AnthonySteele** [comment]
Well spotted!
Yes, [this line should be deleted](https://github.com/NuKeeperDotNet/NuKeeper/blob/master/Directory.Build.props#L4) as it's not the same of all projectss any more.

**Comment by Maxwe11** [comment]
Changing `Directory.Build.props` din't help.

**Comment by AnthonySteele** [comment]
It's a different error now:

```
2020-02-03T11:42:18.1033737Z /opt/hostedtoolcache/dotnet/sdk/3.1.101/NuGet.targets(123,5): 
error : Value cannot be null. (Parameter 'folderName') [/home/vsts/work/1/s/NuKeeper.sln]
```

File: .travis.yml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 language: csharp
 mono: 5.18.1
 dist: xenial
-dotnet: 2.2.202
+dotnet: 3.1.101
 script:
-  - dotnet build -c Release NuKeeper.sln /m:1
+  - dotnet build -c Release --framework netcoreapp3.0 NuKeeper.sln /m:1
```
Comment (slang25) [inline .travis.yml]: If you remove the framework arg I think it will work

**Comment by Maxwe11** [comment]
@slang25  @AnthonySteele  thank you guys

**Comment by AnthonySteele** [comment]
The Travis build has been [timing out](https://travis-ci.org/NuKeeperDotNet/NuKeeper/builds/645868080?utm_medium=notification&utm_source=github_status), looks like [it was failing this "ten  minutes without output" timeout](https://docs.travis-ci.com/user/customizing-the-build/#build-timeouts).

The latest run succeeded, so maybe it's an intermittent / timing issue? Is it possible to either increase the timeout or just generate more output?

Since it's green, I assume that it's good to merge now?

**Pull request merged**